release-prep/action.yml

```diff
@@ -0,0 +1,60 @@
+name: 'Release Prep'
+description: 'Prepares a release by comparing base and head refs'
+
+inputs:
+  base_ref:
+    description: 'The base reference to compare against'
+    required: true
+  head_ref:
+    description: 'The head reference to compare'
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Validate Environment Variables
+      shell: bash
+      run: |
+        required_vars=(
+          "ASANA_PAT"
+          "ASANA_PROJECT_ID"
+          "ASANA_SECTION_ID"
+          "ATLASSIAN_API_TOKEN"
+          "ATLASSIAN_EMAIL"
+          "ATLASSIAN_URL"
+          "CONFLUENCE_RELEASES_SPACE_KEY"
+          "CONFLUENCE_RELEASES_VERSIONS_PARENT_PAGE_ID"
+          "GITHUB_ORG"
+          "GITHUB_PAT"
+          "GITHUB_REPO"
+        )
+        
+        missing_vars=()
+        for var in "${required_vars[@]}"; do
+          if [ -z "${!var}" ]; then
+            missing_vars+=("$var")
+          fi
+        done
+        
+        if [ ${#missing_vars[@]} -ne 0 ]; then
+          echo "Error: The following required environment variables are not set:"
+          printf '%s\n' "${missing_vars[@]}"
+          exit 1
+        fi
+
+    - uses: ruby/setup-ruby@v1
+      with:
+        ruby-version: 'latest'
+        bundler: 'latest'
+        bundler-cache: true
+        working-directory: './release-prep'
+
+    - name: Install dependencies
+      shell: bash
+      working-directory: './release-prep'
+      run: bundle install
+
+    - name: Run Release Prep
+      shell: bash
+      working-directory: './release-prep'
+      run: bundle exec ./bin/release_prep --base-ref="${{ inputs.base_ref }}" --head-ref="${{ inputs.head_ref }}" 
```

Comment:
We may consider `repo` being a passed in arg instead of a ENV. May be less important in the GH actions context but IMO its clearer when invoking as a command. Making it less likely that someone accidentally invokes against the wrong repo.
`./bin/release_prep --base-ref v10.0-rc.1 --head-ref v11.0-rc-2 --repo wharf`